eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/CDDL.hs

```diff
@@ -75,21 +80,15 @@ operational_cert =
       , "sigma" ==> signature
       ]
 
--- TODO Replace with the following once
--- https://github.com/input-output-hk/cuddle/issues/29 is addressed in cuddle.
---
--- next_major_protocol_version :: Rule
--- next_major_protocol_version = "next_major_protocol_version" =:= (10 :: Integer)
-next_major_protocol_version :: Integer
-next_major_protocol_version = 3
+major_protocol_version :: forall era. Era era => Rule
+major_protocol_version =
+  "major_protocol_version"
+    =:= (getVersion @Integer (eraProtVerLow @ByronEra) ... succ (getVersion @Integer (eraProtVerHigh @era)))
```

Comment:
Is it the case that on any era the protocol version provided can be the one of any of the previous eras too? Why isn't this:

```suggestion
    =:= (getVersion @Integer (eraProtVerLow @era) ... succ (getVersion @Integer (eraProtVerHigh @era)))
```
?

Reply:
FTR. You would be able to submit a transaction that proposes to upgrade to an incompatible protocol version (eg. major protocol version `1` in Babbage), but such transaction will be rejected due to Ledger rules, not the decoders.  CDDL spec is only concerned with serializaton, not the rules.